### PR TITLE
Deprecate cpuset field in v1.ScyllaCluster

### LINKED ISSUE #508: CpuSet research
Despite following comment around enabling cpuset for Scylla I tried setting in on environment with non-static CPU policy and Scylla did not crash even when cpuset was counting CPUs not allocated by the kubelet. 
https://github.com/scylladb/scylla-operator/blob/74caca42dd9d27650e9fe2ce8079b815f2835928/pkg/controllers/sidecar/config/config.go#L235-L236

Maybe we should enable it by default and depracate switch in CRD. 
Before doing that, we should check whether setting it always affect anyhow performance or stability of the cluster.

### PR DESCRIPTION
**Pull request by zimnx**
**Description of your changes:**

This deprecates `scyllacluster.spec.cpuset` field and removes mentions of it from examples and documentation.

**Which issue is resolved by this Pull Request:**
Resolves #508

### CODE REVIEW

File: pkg/api/scylla/v1/types_cluster.go [hunk 1]
```diff
@@ -64,6 +64,7 @@ type ScyllaClusterSpec struct {
 	DeveloperMode bool `json:"developerMode,omitempty"`
 
 	// cpuset determines if the cluster will use cpu-pinning for max performance.
```
Comment (tnozicka) [inline pkg/api/scylla/v1/types_cluster.go:66]:
```suggestion
	// cpuset determines if the cluster will use cpu-pinning.
```
if it doesn't matter for performance I'd reword it as well

Comment (zimnx) [reply pkg/api/scylla/v1/types_cluster.go]: done

**Review by tnozicka — COMMENTED** [review]
/approve
/lgtm
thansk
